File: Aulas/Aula_48_3008/aula48-2/src/app/models/jogo.ts
export class Jogo{
  public id!: number;
  public nome!: string;
  public descricao!: string;
  public preco!: number;
  public imagem!: string;
  public site!: string;
  public esta_favoritado!: boolean;

  constructor(id: number, nome: string, descricao: string, preco: number, imagem: string, site: string, esta_favoritado: boolean){
    this.id = id;
    this.nome = nome;
    this.descricao = descricao;
    this.preco = preco;
    this.imagem = imagem;
    this.site = site;
    this.esta_favoritado = this.esta_favoritado;
  }
}

File: Aulas/Aula_48_3008/aula48-1/src/main/java/com/programando/aula481/model/Jogo.java
package com.programando.aula481.model;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Jogo {
    private int id;
    private String nome;
    private String descricao;
    private double preco;
    private String imagem;
    private String site;
    @JsonProperty("esta_favoritado")
    private boolean estaFavoritado;

    public Jogo() {}

    public Jogo(int id, String nome, String descricao, double preco, String imagem,String site, boolean estaFavoritado) {
        this.id = id;
        this.nome = nome;
        this.descricao = descricao;
        this.preco = preco;
        this.imagem = imagem;
        this.site = site;
        this.estaFavoritado = estaFavoritado;
    }

    public String getSite() {
        return site;
    }

    public void setSite(String site) {
        this.site = site;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public String getImagem() {
        return imagem;
    }

    public void setImagem(String imagem) {
        this.imagem = imagem;
    }

    public double getPreco() {
        return preco;
    }

    public void setPreco(double preco) {
        this.preco = preco;
    }

    public boolean isEstaFavoritado() {
        return estaFavoritado;
    }

    public void setEstaFavoritado(boolean estaFavoritado) {
        this.estaFavoritado = estaFavoritado;
    }
}
